# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/glibc/hotfix-x86-Centaur.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- glibc-2.39/sysdeps/x86/dl-cacheinfo.h.vanilla	2024-02-17 22:42:37.049821897 +0100
+++ glibc-2.39/sysdeps/x86/dl-cacheinfo.h	2024-02-17 22:56:39.455740728 +0100
@@ -837,7 +837,7 @@
 
       get_common_cache_info (&shared, &shared_per_thread, &threads, core);
     }
-  else if (cpu_features->basic.kind == arch_kind_zhaoxin)
+  else if (cpu_features->basic.kind == arch_kind_zhaoxin && cpu_features->basic.max_cpuid >= 4)
     {
       data = handle_zhaoxin (_SC_LEVEL1_DCACHE_SIZE);
       core = handle_zhaoxin (_SC_LEVEL2_CACHE_SIZE);
